modules/weaviate/weaviate_test.go

```diff
@@ -74,3 +79,109 @@ func TestWeaviate(t *testing.T) {
 		}
 	})
 }
+
+func TestWeaviateWithClient(t *testing.T) {
+	ctx := context.Background()
+
+	container, err := weaviate.RunContainer(ctx, testcontainers.WithImage(weaviateTestImage))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clean up the container after the test is complete
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err)
+		}
+	})
+
+	t.Run("check", func(tt *testing.T) {
+		httpScheme, httpHost, err := container.HttpHostAddress(ctx)
+		if err != nil {
+			tt.Fatal(err)
+		}
+		grpcHost, err := container.GrpcHostAddress(ctx)
+		if err != nil {
+			tt.Fatal(err)
+		}
+		config := wvt.Config{Scheme: httpScheme, Host: httpHost, GrpcConfig: &wvtgrpc.Config{Host: grpcHost}}
+		client, err := wvt.NewClient(config)
+		if err != nil {
+			tt.Fatal(err)
+		}
+		meta, err := client.Misc().MetaGetter().Do(ctx)
+		if err != nil {
+			tt.Fatal(err)
+		}
+
+		if meta == nil || meta.Version == "" {
+			tt.Fatal("failed to get /v1/meta response")
+		}
+	})
+}
+
+func TestWeaviateWithClientWithModules(t *testing.T) {
```

Comment:
Hey @antas-marcin isn't this test the same one as the testable example?

I always find difficult to choose if writing one or the other, but in the end I try (not always with success) to have only one of them. If the main purpose of this code is to demonstrate how to bootstrap the client, I'd choose a testable example, else a test is fine.

Because both of them takes ~80 seconds, I think it's not meaningful for the total build time, so this comment is not a blocker for the merge. I'd appreciate your thoughts regarding choosing one or the other, or even both.

Thanks!

Reply:
@mdelapenya that's a good point! I was also thinking about this, but I went with leaving both of those tests, so let me remove this test and just leave 1 in examples.